src/parser.ts

```diff
@@ -0,0 +1,272 @@
+import fs from 'fs';
+import os from 'os';
+
+export const VALID_BASE_FORMATS = [
+  'text',
+  'phone_number',
+  'email',
+  'url',
+  'sid',
+  'integer',
+  'number',
+  'secret',
+] as const;
+
+export type BaseVariableFormat = typeof VALID_BASE_FORMATS[number];
+export type ListFormat = string;
+export type NestedListFormat = string;
+export type VariableFormat = BaseVariableFormat | ListFormat | NestedListFormat;
+
+export type VariableDeclaration = {
+  key: string;
+  required: boolean;
+  format: VariableFormat;
+  description: string | null;
```

Comment:
why the distinction of adding `null` to only these 3 variables? Whats different about `key` that doesn't require it?

Reply:
If we are backwards compatible to existing .env files a description or any of the other parts might not be there but the key will always have to be there